soc/arm/st_stm32/stm32wba/power.c

```diff
@@ -60,35 +75,80 @@ void set_mode_stop(uint8_t substate_id)
 	}
 }
 
+#if defined(CONFIG_PM_S2RAM)
+static int suspend_to_ram(void)
+{
+	LL_LPM_EnableDeepSleep();
+
+	while (LL_PWR_IsActiveFlag_ACTVOS() == 0) {
+	}
+
+	/* Select mode entry : WFE or WFI and enter the CPU selected mode */
+	k_cpu_idle();
+
+	return 0;
+}
+
+void set_mode_suspend_to_ram(uint8_t substate_id)
+{
+	ARG_UNUSED(substate_id);
+
+	/* Enable SRAM full retention */
+	LL_PWR_SetSRAM1SBRetention(LL_PWR_SRAM1_SB_FULL_RETENTION);
+	LL_PWR_SetSRAM2SBRetention(LL_PWR_SRAM2_SB_FULL_RETENTION);
+
+	/* Enable RTC wakeup
+	 * This configures an internal pin that generates an event to wakeup the system
+	 */
+	LL_PWR_EnableWakeUpPin(LL_PWR_WAKEUP_PIN7);
+	LL_PWR_SetWakeUpPinSignal3Selection(LL_PWR_WAKEUP_PIN7);
+
+	/* Clear flags */
+	LL_PWR_ClearFlag_SB();
+	LL_PWR_ClearFlag_WU();
+	LL_RCC_ClearResetFlags();
+
+	disable_cache();
+
+	/* Select standby mode */
+	LL_PWR_SetPowerMode(LL_PWR_MODE_STANDBY);
+
+	/* Save context and enter Standby mode */
+	arch_pm_s2ram_suspend(suspend_to_ram);
+
+	/* System is restored at this point after wake up */
```

Comment:
Let's be extra-explicit:
```suggestion
	/* Execution is resumed at this point after wake up */
```